// util-tiempo/dist/lib/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getNextTime = exports.getTime = exports.getMs = exports.getFormatDate = exports.getDate = exports.getCompareDate = exports.get = void 0;
const get_1 = __importDefault(require("./get"));
exports.get = get_1.default;
const getCompareDate_1 = __importDefault(require("./getCompareDate"));
exports.getCompareDate = getCompareDate_1.default;
const getDate_1 = __importDefault(require("./getDate"));
exports.getDate = getDate_1.default;
const getFormatDate_1 = __importDefault(require("./getFormatDate"));
exports.getFormatDate = getFormatDate_1.default;
const getMs_1 = __importDefault(require("./getMs"));
exports.getMs = getMs_1.default;
const getTime_1 = __importDefault(require("./getTime"));
exports.getTime = getTime_1.default;
const getNextTime_1 = __importDefault(require("./getNextTime"));
exports.getNextTime = getNextTime_1.default;
